dlp/api/Snippets/DeidentifyDataStoredInCloudStorage.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// [START dlp_deidentify_cloud_storage]
+
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Dlp.V2;
+using System.Linq;
+
+public class DeidentifyDataStoredInCloudStorage
+{
+    public static DlpJob Deidentify(
+        string projectId,
+        string gcsInputPath,
+        string unstructuredDeidentifyTemplatePath,
+        string structuredDeidentifyTemplatePath,
+        string imageRedactionTemplatePath,
+        string gcsOutputPath,
+        string datasetId,
+        string tableId)
+    {
+        // Instantiate the client.
+        var dlp = DlpServiceClient.Create();
+
+        //Construct the storage config by specifying the input directory.
+        var storageConfig = new StorageConfig
+        {
+            CloudStorageOptions = new CloudStorageOptions
+            {
+                FileSet = new CloudStorageOptions.Types.FileSet
+                {
+                    Url = gcsInputPath
+                }
+            }
+        };
+
+        // Construct the inspect config by specifying the type of info to be inspected.
+        var inspectConfig = new InspectConfig
+        {
+            InfoTypes =
+            {
+                new InfoType[]
+                {
+                    new InfoType { Name = "PERSON_NAME" },
+                    new InfoType { Name = "EMAIL_ADDRESS" }
+                }
+            },
+            IncludeQuote = true
+        };
+
+        // Construct the actions to take after the inspection portion of the job is completed.
+        // Outputs will be stored in specified GCS output path and specified table.
```

Comment:
The de-identified files will be written to the the GCS bucket path specified in gcsOutputPath
The details of the transformations performed will be written to BQ table specified (datasetId, tableId).
We should document this clearly. Please update comments.

Reply:
Done. Thanks